cli/azd/cmd/provision.go

```diff
@@ -129,12 +137,30 @@ func (p *provisionAction) Run(ctx context.Context) (*actions.ActionResult, error
 		Project: p.projectConfig,
 	}
 
+	reservedParameters := []string{
+		"environmentName",
+	}
 	err := p.projectConfig.Invoke(ctx, project.ProjectEventProvision, projectEventArgs, func() error {
 		deploymentPlan, err := p.provisionManager.Plan(ctx)
 		if err != nil {
 			return fmt.Errorf("planning deployment: %w", err)
 		}
 
+		// Specified parameters override other parameters.
+		parameters := deploymentPlan.Deployment.Parameters
```

Comment:
I'd like to see the custom/override integrated directly into our provisioning manager vs this only being available at the command level.

Or - expose a new method on the `deploymentPlan` to set the overrides instead of us having to augment the provisioning manager/providers contract.

Reply:
Yes - We need to be passing these values into `Plan` and using them there.  For bicep it doesn't much matter today (but will once we want to support something like `--preview` which is on the roadmap), but for terraform it is very important we do so, because the `tf plan` operation needs to see the correct value of every parameter (both so it doesn't prompt for any values that were not specified in the tf vars file and also because these values will influence the generated plan).

Reply:
This is why I started a thread in the Teams channel about why there are even two structs with parameters to begin with. It doesn't feel like a good design, and from what I can tell the "global" set of parameters aren't even used after initial parsing. This seems like a bug or an oversight and I want to discuss further. I think we should use that set of parameters ubiquitously. Please see my recent post in our Teams channel.

Reply:
I've @mentioned both of you in the thread. May be faster to resolve this there and include the whole team. I personally feel how parameters are handled now seems wrong and happy to revamp them here if needed. Could also be a separate PR just to keep them focused (better for future troubleshooting or even reversion).